### PR TITLE
Highlighting Owned Token in Attendees List

### PR DESCRIPTION
**Pull request by quadcopterman**
I created an Adapter with a new getView() function. This function is then called when the tokens are displayed in a rollcall, changed how our token is displayed (to highlight only our token)
This solves issue #1776

### CODE REVIEW

**Review by Kaz-ookid — COMMENTED** [review]
LGTM!

It does not seem like this change requires testing, but you should make sure of it and ask a TA first :)

I have been successfully testing it and saw no issue. Maybe we'll keep an eye on it during the next global meeting!

Also, a small note on the PR description - the title "Fix issue #1776" could be more descriptive. How about "Highlighting Owned Token in Attendees List"? It's always good to be clear about what the PR does. Plus, mentioning the issue number as #1776 in the description creates a clickable link to the issue, which is super handy for quick navigation.

Keep up the great work!

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/event/rollcall/RollCallArrayAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+) : ArrayAdapter<String>(context, layout, attendeesList) {
+
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
+        val view = super.getView(position, convertView, parent)
+
+        //highlights our token in the list
+        val currentToken = getItem(position)
+        if (myToken != null && currentToken == myToken.publicKey.encoded) {
+            (view as TextView).setTypeface(null, Typeface.BOLD)
+            view.setTextColor(Color.BLUE)
```
[the remark this answers is not in the document]
Comment (matteosz) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/event/rollcall/RollCallArrayAdapter.kt:26]: Yes absolutely agree with @Kaz-ookid, avoid using constants for this kind of things, as very often there are reusable components, and use resources instead

**Review by matteosz — CHANGES_REQUESTED** [review]
LGTM, just 2 comments about using resources and testing

**Comment by matteosz** [comment]
> LGTM!
> 
> It does not seem like this change requires testing, but you should make sure of it and ask a TA first :)
> 
> I have been successfully manually testing it and saw no issue. Maybe we'll keep an eye on it during the next global meeting!
> 
> Also, a small note on the PR description - the title "Fix issue #1776" could be more descriptive. How about "Highlighting Owned Token in Attendees List"? It's always good to be clear about what the PR does. Plus, mentioning the issue number as #1776 in the description creates a clickable link to the issue, which is super handy for quick navigation.
> 
> Also, make sure to mark this PR as "ready for review" when you think you are done ! 👍
> 
> Keep up the great work!

Actually, I'd say that all the changes, even small ones, need validation. Testing this seems also straightforward (create a roll call with 2 attendees and verify that own token is highlighted and the other not) so I'd suggest to go for it

**Comment by quadcopterman** [comment]
Thanks for your comments, I'll change the mentioned points.
For manual testing, I can't do a roll call for the moment as my emulator doesn't have access to my camera for some reason... I will try to solve that today and then do the requested changes

**Comment by matteosz** [comment]
> Thanks for your comments, I'll change the mentioned points. For manual testing, I can't do a roll call for the moment as my emulator doesn't have access to my camera for some reason... I will try to solve that today and then do the requested changes

 Actually I'd suggest unit tests rather than manual testing, also because the code coverage is not above threshold

File: fe2-android/app/src/test/ui/robolectric/com/github/dedis/popstellar/ui/lao/event/rollcall/RollCallArrayAdapterTest.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Mock
+    private lateinit var mockView: View
+
+    private lateinit var adapter: RollCallArrayAdapter
+
+    val context = ApplicationProvider.getApplicationContext<Context>()
+
+    private val MY_PRIVATE_KEY =
+            Utils.hexToBytes("3b28b4ab2fe355a13d7b24f90816ff0676f7978bf462fc84f1d5d948b119ec66")
```
Comment (matteosz) [inline fe2-android/app/src/test/ui/robolectric/com/github/dedis/popstellar/ui/lao/event/rollcall/RollCallArrayAdapterTest.kt:31]: It's fine for me having this hardcoded, as it doesn't impact the test coverage, but we have in the codebase a random generator for keys that we use in test, just pointing this for future work as it's gonna save you some time

**Review by matteosz — APPROVED** [review]
It's fine by me, but the test is just a unit test for the adapter, which is obviously okay. I'd have liked to have at least 1 integration test that mimics the users' behaviour in the UI though, so adding a test in the RollCallFragment test, where you create a rollcall there and verify everything works (correct token is highlighted), in that case you could have avoided having a specific unit test for the adapter. But it's just my personal preference, it kinda works the same either way :)

File: fe2-android/app/src/test/ui/robolectric/com/github/dedis/popstellar/ui/lao/event/rollcall/RollCallArrayAdapterTest.kt (hunk 2 above)
Comment (quadcopterman) [reply fe2-android/app/src/test/ui/robolectric/com/github/dedis/popstellar/ui/lao/event/rollcall/RollCallArrayAdapterTest.kt:31]: Oh ok, thanks for the information

**Comment by Kaz-ookid** [comment]
LGTM!

**Pull request merged**